### DISCUSSION TITLE
Help needed for BOM-VEX link

### DISCUSSION THREAD

**Discussion by pecros** [DependencyTrack/dependency-track#1921] [Q&A] [answered]
Hello,
I couldn't succeed to add vulnerabilities to a test project with a VEX. 
Would someone have a simple BOM and VEX test files that I can import in dependency-track as an example ? 
Thank you very much.
Best regards

**Comment by syalioune** [comment, marked as answer]
Hello @pecros 

You maybe have a wrong assumption regarding VEX usage.
My understanding as per [cyclonedx vex capability](https://cyclonedx.org/capabilities/vex/) and DT codebase is that you would use VEX to convey audit/exploitability data to an **existing** vulnerability attached to a SBOM component.
You cannot use VEX (at least in DT) to import new vulnerabilities. The vulnerability should already be known by DT and linked to the project/component.

PFA [bom.txt](https://github.com/DependencyTrack/dependency-track/files/9436963/bom.txt) and [vex.txt](https://github.com/DependencyTrack/dependency-track/files/9436964/vex.txt) allowing to set CVE `CVE-2022-28391` as `exploitable`.
They should be uploaded in the correct order : SBOM first, DT analysis complete and vuln identified, then VEX.

Note that in the snippet below extracted from attached VEX, the affects[0].ref value refer to the metadata.component bom-ref (i.e. the project itself) so the analysis state and detail apply to all project components vulnerable to `CVE-2022-28391`. You can also refer to specific sub-component to scope the analysis. 

```json
    {
      "bom-ref" : "e32e057a-114b-404b-816d-fbeb99b7d260",
      "id" : "CVE-2022-28391",
      "source" : {
        "name" : "NVD",
        "url" : "https://nvd.nist.gov/"
      },
      "ratings" : [
        {
          "source" : {
            "name" : "NVD",
            "url" : "https://nvd.nist.gov/"
          },
          "score" : 6.8,
          "severity" : "medium",
          "method" : "CVSSv2",
          "vector" : "(AV:N/AC:M/Au:N/C:P/I:P/A:P)"
        },
        {
          "source" : {
            "name" : "NVD",
            "url" : "https://nvd.nist.gov/"
          },
          "score" : 8.8,
          "severity" : "high",
          "method" : "CVSSv3",
          "vector" : "CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H"
        }
      ],
      "description" : "BusyBox through 1.35.0 allows remote attackers to execute arbitrary code if netstat is used to print a DNS PTR record's value to a VT compatible terminal. Alternatively, the attacker could choose to change the terminal's colors.",
      "published" : "2022-04-03T21:15:00Z",
      "updated" : "2022-08-11T18:44:00Z",
      "analysis" : {
        "state" : "exploitable",
        "detail" : "Detailed explanation"
      },
      "affects" : [
        {
          "ref" : "7f2ee811-6b35-4c24-83ec-605d7939005c"
        }
      ]
    }
```

**Reply by pecros** [reply]
Hello,
It works, thank you very much for the explanation and the example!
Best

**Reply by JoergBruenner** [reply]
Hello @syalioune,

thank you for the good example. How can I supress the vulnerabilities while applying a VEX?

Thank you Jörg

**Reply by syalioune** [reply]
Hello @JoergBruenner 

As long as you set the state to `false_positive`, `not_affected` or `resolved` in the analysis, the vulnerability will be suppressed.

**Comment by JoergBruenner** [comment]
Thank you  :-)Von meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: syalioune ***@***.***> Datum: 08.02.23  21:41  (GMT+01:00) An: DependencyTrack/dependency-track ***@***.***> Cc: JoergBruenner ***@***.***>, Mention ***@***.***> Betreff: Re: [DependencyTrack/dependency-track] Help needed for BOM-VEX link
  (Discussion #1921) 
Hello @JoergBruenner
As long as you set the state to false_positive, not_affected or resolved in the analysis, the vulnerability will be suppressed.
